src/parse.cpp

```diff
@@ -1,10 +1,50 @@
 #include "parse.hpp"
-
 #include "lexer.hpp"
+#include <iostream>
+#include <stdexcept>
+
+std::unique_ptr<FunctionAST> parseFunction(TokenStream &stream);
+std::vector<std::string> parseVarInput(TokenStream &stream);
 
-// TODO: This return type might need to change
-std::unique_ptr<FunctionAST> parse(std::vector<Token> tokens)
+std::unique_ptr<FunctionAST> parse(TokenStream &stream)
 {
-    // TODO: Implement parsing
+    auto tok = stream.next();
+    if (tok.getkey() == "KEYWORD" && tok.getvalue() == "FUNCTION")
+    {
+        std::cout << "Parsing function..." << std::endl;
```

Comment:
```suggestion
        std::cerr << "Parsing function..." << std::endl;
```

In general, things that are not the desired output of the program should not be printed to stdout, they should go to stderr. However, this log message should be removed before merging.